src/helpers/user_agent.js

```diff
@@ -1,3 +1,11 @@
 const { userAgent } = window.navigator
 
-export const isHotwireNativeApp = /bridge-components: \[.+\]/.test(userAgent)
+export function appSupportsBridgeComponent(component) {
+  const supportedComponents = userAgent.match(/bridge-components: \[(.*?)\]/)
+
+  if (supportedComponents) {
+    return supportedComponents[1].split(" ").includes(component)
```

Comment:
Wouldn't an empty bridge-components array cause this to be undefined? For example, `HotwireNative iOS/1.1.1 (build 251) bridge-components: []`.

Reply:
Good question, but en empty array is handled fine here. The regex matches to an empty string and evaluates fine:

```
> "Hotwire Native; bridge-components: [];".match(/bridge-components: \[(.*?)\]/)[1].split(" ").includes("my-component")

false
```